src/telemetry_layer.rs

```diff
@@ -0,0 +1,106 @@
+use std::{env::var, path::PathBuf};
+use tracing::{
+    span::{self, Id, Record},
+    Event,
+};
+use tracing_appender::non_blocking::{NonBlocking, WorkerGuard};
+use tracing_subscriber::layer::Context;
+use tracing_subscriber::{
+    fmt::{format::DefaultFields, Layer},
+    Layer as LayerTrait, Registry,
+};
+
+use crate::{
+    errors::TelemetryError, telemetry_config, telemetry_formatter::TelemetryFormatter, Result,
+};
+
+/// A `tracing` `Layer` to generate telemetry to be later consumed into InfluxDB
+///
+/// This `tracing` `Layer` formats telemetry and stores the output in a known
+/// location ($FUEL_HOME/tmp/<crate>.telemetry), ready for ingestion by an
+/// InfluxDB collector.
+pub struct TelemetryLayer {
+    pub __inner: Layer<Registry, DefaultFields, TelemetryFormatter, NonBlocking>,
```

Comment:
Dont think this should be `pub` if prefixed using `__`; instead use a getter function; or remove the `__` prefix?
```suggestion
    __inner: Layer<Registry, DefaultFields, TelemetryFormatter, NonBlocking>,
```

Reply:
Yeah, this one was tricky given that `fuel-telemetry-macros` needs mutable access, and every option doesn't look that nice. I'm putting this in the LATER basket and will come back to it, but for now I'll rename to `inner_layer` and keep it public - 47e9a72ca8